frontend/app/components/SpotifyPlayer.tsx

```diff
@@ -0,0 +1,803 @@
+'use client';
+
+import { useEffect, useState, useRef, useCallback } from 'react';
+import { createClient } from '../utils/supabase/client';
+
+declare global {
+  interface Window {
+    Spotify: any;
+    onSpotifyWebPlaybackSDKReady: () => void;
+  }
+}
+
+async function transferPlayback(deviceId: string, accessToken: string) {
+  const response = await fetch('https://api.spotify.com/v1/me/player', {
+    method: 'PUT',
+    headers: {
+      'Authorization': `Bearer ${accessToken}`,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      device_ids: [deviceId],
+      play: true
+    })
+  });
+
+  if (!response.ok) {
+    throw new Error('Failed to transfer playback');
+  }
+}
+
+interface SpotifyArtist {
+  name: string;
+  uri: string;
+}
+
+interface SpotifyTrack {
+  name: string;
+  uri: string;
+  album: {
+    images: { url: string }[];
+  };
+  artists: SpotifyArtist[];
+}
+
+interface QueueState {
```

Comment:
This is going to have to move to the backend - we're also not going to be hardcoding prevTrack,prevTrack2 or nextTrack, nextTrack2 etc..

I just pushed up in 8128b559983a9dc21c006e4a878af8926d4df91a some [new types](https://github.com/theonlyjohnny/vibe-jockey/blob/8128b559983a9dc21c006e4a878af8926d4df91a/frontend/app/types/song-queue.ts) which we'll need to ingest in the next iteration of the player UI